### PR TITLE
feat: admin table functionality

### PR DESCRIPTION
**Pull request by mrkhunt**
**Features added:**
- [x] Table auto updates on save/delete users
- [x] Username, email, password, status, email verified, paid member, payment method, verifier, location are all editable fields for each user.
- [x] Pagination:
  - [x] Index by +/- 1 page
  - [x] jump to start/end page
  - [x] display total rows
  - [x] change rows/page
- [x] Admin is able to perform fuzzy search on all fields
- [x] Reusable table with User Types
  - [x] `TableMeta` includes arbitrary functions accessible across the table
  - [x] `CellMeta` includes arbitrary functions accessible across the column
  - [x] Can extend cell type (currently includes, `TableCell` and `EditCell` for displaying and editing data respectively)

**UI changes:**
- [x] change `UserFormCard` to be responsive
- [x] Username remains sticky to the left side
- [x] Edit Actions remains sticky to the right side
- [x] Table adapts to different screen size (removed sticky columns for medium and smaller screens)

**Refactoring done:**
- [x] Routing API calls for following
  - [x] fetchingUsers
  - [x] creatingUser
  - [x] deletingUser
  - [x] editingUser
 - [x] Admin Page:
  - [x] Table component handles all logic instead of admin page

### CODE REVIEW

**Comment by dscwebdev** [comment]
I noticed that the feature to autofill with the admin's name when we set has paid to true is not working, neither is it clearing when hasPaid is set to false. I've tried debugging and everything behaves as expected except `setEditFormData`, meaning all the logic in `TableCell` works fine but its just not updating.

**Comment by mrkhunt** [comment]
Made some updates:

- fetch email properly
- create user is fixed
- hasPaid updates properly
  - if the field is True, update adminName in verifier
  - if changed to False, update adminName to empty

File: components/tables/TableCell.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      }
+    } else {
+      tableMeta.setEditFormData((prevData: User | null) =>
+        prevData ? { ...prevData, [column.id]: newValue } : null,
+      );
+      console.log("Edit Form Data:", tableMeta.editFormData);
+    }
+  };
+
+  const user = row.original as User;
```
Comment (dscwebdev) [inline components/tables/TableCell.tsx:80]: This variable is not used, maybe we could remove it ? I didn't remove it myself incase it relates to the next comment.

**Review by dscwebdev — APPROVED** [review]
There is a tiny issue still with the verifier column. When you edit, a user and set his/her hasPaid to something different and then cancel the changes, the verifier column does not return to its original value.

Other than that, it LGTM! @xinhuey could you take a look as well?

File: components/tables/TableCell.tsx (hunk 1 above)
Comment (mrkhunt) [reply components/tables/TableCell.tsx]: yep unused - removed

**Comment by xinhuey** [comment]
Great job on this ticket. Looks good to me as well, go ahead and merge it!

**Review by xinhuey — APPROVED** [review]

**Pull request merged**